sdk/messaging/azservicebus/messageSettler.go

```diff
@@ -24,70 +39,79 @@ func (s *messageSettler) useManagementLink(m *ReceivedMessage, linkRevision uint
 		m.linkRevision != linkRevision
 }
 
-// CompleteMessage completes a message, deleting it from the queue or subscription.
-func (s *messageSettler) CompleteMessage(ctx context.Context, message *ReceivedMessage) error {
+func (s *messageSettler) settleWithRetries(ctx context.Context, message *ReceivedMessage, settleFn func(receiver internal.AMQPReceiver, mgmt internal.MgmtClient, linkRevision uint64) error) error {
 	if s == nil {
 		return errReceiveAndDeleteReceiver
 	}
 
-	_, _, mgmt, linkRevision, err := s.links.Get(ctx)
+	retrier := s.baseRetrier.Copy()
+	var lastErr error
 
-	if err != nil {
-		return err
-	}
+	for retrier.Try(ctx) {
+		var receiver internal.AMQPReceiver
+		var mgmt internal.MgmtClient
+		var linkRevision uint64
+
+		_, receiver, mgmt, linkRevision, lastErr = s.links.Get(ctx)
+
+		if lastErr != nil {
+			_ = s.links.RecoverIfNeeded(ctx, lastErr)
```

Comment:
If we always treat RecoverIfNeeded as no-fail perhaps it would be better to have it not return an error (I know that feels weird).

Reply:
Yeah, I was thinking that too. I have post-beta-1 plans where I do want to make usage of it more, but not yet in this one.